### DISCUSSION TITLE
generateStaticParams working but not showing html+json files in build

### DISCUSSION THREAD

**Discussion by Mohammadtahirlashari** [vercel/next.js#52472] [Help]
### Summary

![image](https://github.com/vercel/next.js/assets/61346951/85263d5d-f836-4e7a-a57e-a10aa57a80ef)
I have created SSG with data from API. It is created with the message **(SSG) automatically generated as static HTML + JSON (uses getStaticProps)** but there is no **html+json** files in the server build folder.

I am using NextJS 13.4.9 version.

My code is 
![image](https://github.com/vercel/next.js/assets/61346951/765a0b9a-cbe3-47ad-9ce1-cbe0fb33bab6)

**Comment by icyJoseph** [comment]
Hi,

For a page with path `app/blog/[slug]/page.tsx`, for example, you'd need to return an list of objects with a `slug` key:

```ts
export async function generateStaticParams() {
  const posts = await fetch('https://.../posts').then((res) => res.json())
 
  return posts.map((post) => ({
    slug: post.slug,
  }))
}
```

**_Which is different than what GetStaticPaths required_**, which was an object with a key paths, which in turn had a key called params, which finally contained the URL slots.

So that should actually take care of the problem that you don't have any pages.

To be precise, you need to return:

```js
export async function generateStaticParams() {
  const data = await fetch(/* endpoint */);
  const posts = await data.json()
 
  return posts.map((post) => ({
    id: `${post.id}`,
  }))
}
```

On to the HTML+JSON bit, that is the message for Pages Dir, that has not been updated yet. In your case, your Id page is a Server Component, and it has no interactivity ('use client' is not used), so it'll be flushed to pure HTML, and some RSC data that'll be used to stitch together the client.

When you've fixed the initial problem, you could use https://github.com/jherr/njs13-n-plus-one/blob/main/tester/index.js to see a measure of the HTML and RSC data that's generated. For example for a fresh `create-next-app` with no changes, the production build logs:

```console
            HTML: 5.97 KB - 38.0%
            JSON: 9.75 KB - 62.0%
JSON Client Data: 0.00 KB - 0.0%
```
